x-pack/plugins/maps/public/classes/sources/es_search_source/top_hits/create_source_editor.tsx

```diff
@@ -0,0 +1,148 @@
+    indexPattern: null,
+    geoFields: [],
+    geoFieldName: null,
+    termFields: [],
+    topHitsSplitField: null,
+    topHitsSize: 1,
+  };
+
+  _onIndexPatternSelect = (indexPattern: IndexPattern) => {
+    const geoFields = getGeoFields(indexPattern.fields);
+
+    this.setState(
+      {
+        indexPattern,
+        geoFields,
+        geoFieldName: null,
+        termFields: getTermsFields(indexPattern.fields),
+        topHitsSplitField: null,
+      },
+      () => {
+        if (geoFields.length) {
+          this._onGeoFieldSelect(geoFields[0].name);
+        }
+      }
+    );
+  };
+
+  _onGeoFieldSelect = (geoFieldName?: string) => {
+    this.setState({ geoFieldName: geoFieldName ? geoFieldName : null }, this._previewLayer);
+  };
+
+  _onTopHitsPropChange = ({ propName, value }: OnSourceChangeArgs) => {
+    this.setState(
+      // @ts-expect-error
+      { [propName]: value },
+      this._previewLayer
+    );
+  };
+
+  _previewLayer = () => {
+    const { indexPattern, geoFieldName, topHitsSplitField, topHitsSize } = this.state;
+
+    const tooltipProperties: string[] = [];
+    if (topHitsSplitField) {
+      tooltipProperties.push(topHitsSplitField);
+    }
+    if (indexPattern && indexPattern.timeFieldName) {
+      tooltipProperties.push(indexPattern.timeFieldName);
+    }
+
+    const sourceConfig =
+      indexPattern && geoFieldName && topHitsSplitField
+        ? {
+            indexPatternId: indexPattern.id,
+            geoField: geoFieldName,
+            scalingType: SCALING_TYPES.TOP_HITS,
+            sortField: indexPattern.timeFieldName ? indexPattern.timeFieldName : '',
+            sortOrder: SortDirection.desc,
+            tooltipProperties,
+            topHitsSplitField,
+            topHitsSize,
+          }
+        : null;
+    this.props.onSourceConfigChange(sourceConfig);
+  };
+
+  _renderGeoSelect() {
+    return this.state.indexPattern ? (
+      <GeoFieldSelect
+        value={this.state.geoFieldName ? this.state.geoFieldName : ''}
+        onChange={this._onGeoFieldSelect}
+        geoFields={this.state.geoFields}
+      />
+    ) : null;
+  }
+
+  _renderTopHitsPanel() {
+    if (!this.state.indexPattern || !this.state.indexPattern.id || !this.state.geoFieldName) {
+      return null;
+    }
+
+    return (
+      <TopHitsForm
+        indexPatternId={this.state.indexPattern.id}
+        isColumnCompressed={false}
+        onChange={this._onTopHitsPropChange}
+        termFields={this.state.termFields}
+        topHitsSplitField={this.state.topHitsSplitField}
+        topHitsSize={this.state.topHitsSize}
+      />
+    );
+  }
+
+  render() {
+    return (
+      <EuiPanel>
+        <GeoIndexPatternSelect
+          value={
+            this.state.indexPattern && this.state.indexPattern.id ? this.state.indexPattern.id : ''
+          }
+          onChange={this._onIndexPatternSelect}
+        />
+
+        {this._renderGeoSelect()}
+
+        {this._renderTopHitsPanel()}
```

Comment:
I'd consider adding in the sort-field and sort-direction here as well.

![image](https://user-images.githubusercontent.com/1833023/113180330-9caa1b00-921e-11eb-9be5-b593a46a7d68.png)

^ this is the layer-settings and while more "cluttered", it signals the importance of the sort-field. Without the "sort" there is no "top".  (the "dynamically filter" switch would not be necessary imho).